include/roboteam_ai/treeinterp/TreeProtoType.h

```diff
@@ -0,0 +1,49 @@
+//
+// Created by jesse on 14.10.19.
+//
+#ifndef RTT_TREE_PROTO_TYPE_H
+#define RTT_TREE_PROTO_TYPE_H
+
+#include <include/roboteam_ai/bt/tactics/DefaultTactic.h>
+#include "bt/BehaviorTree.hpp"
+#include "bt/Role.h"
+
+
+
+namespace bt {
+
+    class TreeProtoType {
+    public:
+        TreeProtoType();
+        /**
+         * Creates a Strategy behaviour tree. This tree is created in BTFactory.
+         * @return the behaviour tree that contains this strategy
+         */
+        std::shared_ptr<bt::BehaviorTree> createOffensiveStrategy();
+
+    private:
+        /**
+         * Vector that is used by RobotDealer to determine how to match robots to robot IDs.
+         */
+        std::vector<std::pair<std::string, rtt::ai::robotDealer::RobotType>> robots;
```

Comment:
So basically the only state this object holds is the robotDealer labels, which are never updated. You might want to consider just making this a function

Reply:
I'm not sure I understand what you mean. I think making this vector in the constructor is nice, since it means you always have some default robotdealer config, but if you decide later on you want the strategy but with some other flags you can easily build this functionality this way.

Reply:
I think I understand what you mean now, making everything static bypasses the constructor so I agree that making a method for this is better